src/devices/tuya.ts

```diff
@@ -11447,6 +11447,57 @@ const definitions: DefinitionWithExtend[] = [
             ],
         },
     },
+    {
+        fingerprint: tuya.fingerprint('TS0601', ['_TZE204_fhvdgeuh']),
+        model: 'TS0601',
+        vendor: 'Tuya',
+        description: 'Din rail switch with power monitoring and threshold settings',
+        fromZigbee: [tuya.fz.datapoints, fz.metering],
+        toZigbee: [tuya.tz.datapoints],
+        configure: async (device, coordinatorEndpoint) => {
+            await tuya.configureMagicPacket(device, coordinatorEndpoint);
+            // Required to get the device to start reporting
+            await device.getEndpoint(1).command('manuSpecificTuya', 'dataQuery', {});
+        },
+        exposes: [
+            e.switch().setAccess('state', ea.STATE_SET),
+            e.power(),
+            e.current(),
+            e.voltage(),
+            e.numeric('energy', ea.STATE).withUnit('kWh').withDescription('Sum of consumed energy'),
+            e.numeric('current_temperature', ea.STATE).withUnit('°C').withDescription('Current temperature'),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [16, 'state', tuya.valueConverter.onOff],
+                [1, 'energy', tuya.valueConverter.divideBy100], // Total forward energy
+                [6, null, tuya.valueConverter.phaseVariant2], // Phase A voltage and current
+                [9, 'Fault', tuya.valueConverter.raw],
+                [11, 'Switch prepayment', tuya.valueConverter.raw],
+                [12, 'Clear energy', tuya.valueConverter.raw],
+                // [13, 'Balance energy (kWh)', tuya.valueConverter.raw], // wrong, is always 0
+                [14, 'charge energy', tuya.valueConverter.raw],
+                [15, 'Leakage current', tuya.valueConverter.raw],
+                // [17, 'Alarm set1',tuya.valueConverter.raw], noise
+                // [18, 'Alarm set2',tuya.valueConverter.raw], noise
+                // [101, 'Closing recovery time (Deprecated)',tuya.valueConverter.raw],
+                [102, 'reclosing allowed times', tuya.valueConverter.raw],
+                [103, 'current_temperature', tuya.valueConverter.raw],
+                [104, 'reclosing enable', tuya.valueConverter.raw],
+                [105, 'Timer', tuya.valueConverter.raw],
+                [106, 'Cycle Schedule', tuya.valueConverter.raw],
+                [107, 'reclose recover seconds', tuya.valueConverter.raw],
+                [108, 'Random Timing', tuya.valueConverter.raw],
+                [109, 'Switch Inching', tuya.valueConverter.raw],
+                [119, 'Power-on delay power-on time', tuya.valueConverter.raw],
```

Comment:
Please use snake casing every where

```suggestion
                [119, 'power_on_delay_power_on_time', tuya.valueConverter.raw],
```

Reply:
Done